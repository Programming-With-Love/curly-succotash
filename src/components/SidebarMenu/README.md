# SidebarMenu component

Sidebar menu component.

## Source

    <SidebarMenu visible={visible}  />
