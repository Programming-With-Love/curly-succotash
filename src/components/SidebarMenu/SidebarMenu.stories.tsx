/* tslint:disable no-var-requires */
const withReadme = (require('storybook-readme/with-readme') as any).default
const SidebarMenuReadme = require('./README.md')

import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'
import { SidebarMenu } from './SidebarMenu'

const items = [
  { name: 'Home', path: '/', exact: true, icon: 'home' },
  { name: 'About', path: '/about/', exact: true, icon: 'info circle' },
  { name: 'Blog', path: '/blog/', exact: false, icon: 'newspaper' },
]

const LinkStub: any = (props: any) => (
  <div {...props} onClick={action(props.to.toString())}>
    {props.children}
  </div>
)

storiesOf('SidebarMenu', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(SidebarMenuReadme))
  .add('default', () => {
    const pathname = text('pathname', '/')

    return <SidebarMenu Link={LinkStub} items={items} pathname={pathname} visible />
  })
