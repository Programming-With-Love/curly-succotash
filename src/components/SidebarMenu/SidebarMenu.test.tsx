import { render, configure } from 'enzyme'
import 'jest'
import * as React from 'react'
import { SidebarMenu } from './SidebarMenu'

// Configure enzyme with react 16 adapter
const Adapter: any = require('enzyme-adapter-react-16')
configure({ adapter: new Adapter() })

const items = [
  { name: 'Home', path: '/', exact: true },
  { name: 'About', path: '/about/', exact: true },
  { name: 'Blog', path: '/blog/', exact: false },
]

const LinkStub: any = (props: any) => <div {...props} />

describe('SidebarMenu component', () => {
  it('should render correctly', () => {
    const wrapper = render(<SidebarMenu Link={LinkStub} pathname="/" items={items} visible />)
    expect(wrapper).toMatchSnapshot()
  })
})
