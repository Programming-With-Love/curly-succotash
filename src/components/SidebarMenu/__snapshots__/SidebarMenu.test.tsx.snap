// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenu component should render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ui vertical labeled icon ui slide along left thin visible sidebar menu",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "active item",
          "to": "/",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "aria-hidden": "true",
              "class": "icon",
            },
            "children": Array [],
            "name": "i",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "data": "Home",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "text",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-hidden": undefined,
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-hidden": undefined,
              "class": undefined,
            },
          },
          Object {
            "data": "Home",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "icon",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "item",
            "to": "/about/",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "icon",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "data": "About",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "text",
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
            Object {
              "data": "About",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "icon",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "item",
              "to": "/blog/",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "icon",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "data": "Blog",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "text",
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
              Object {
                "data": "Blog",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                },
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "to": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "to": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "to": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "to": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "to": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "to": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "item",
          "to": "/about/",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "aria-hidden": "true",
              "class": "icon",
            },
            "children": Array [],
            "name": "i",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "data": "About",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "text",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-hidden": undefined,
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-hidden": undefined,
              "class": undefined,
            },
          },
          Object {
            "data": "About",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "icon",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "item",
            "to": "/blog/",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "icon",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "data": "Blog",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "text",
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
            Object {
              "data": "Blog",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "icon",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "to": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "to": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "active item",
            "to": "/",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "icon",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "data": "Home",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "text",
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
            Object {
              "data": "Home",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "icon",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "to": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "to": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "to": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "to": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "item",
          "to": "/blog/",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "aria-hidden": "true",
              "class": "icon",
            },
            "children": Array [],
            "name": "i",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "data": "Blog",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "text",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-hidden": undefined,
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-hidden": undefined,
              "class": undefined,
            },
          },
          Object {
            "data": "Blog",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "icon",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "item",
            "to": "/about/",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "icon",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "data": "About",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "text",
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
            Object {
              "data": "About",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "icon",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "active item",
              "to": "/",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "icon",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "data": "Home",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "text",
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
              Object {
                "data": "Home",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                },
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "to": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "to": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "to": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "to": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "to": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "to": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
